src/sidebar/config/build-settings.js

```diff
@@ -0,0 +1,147 @@
+import { getApiUrl } from './get-api-url';
+import { hostPageConfig } from './host-config';
+import * as postMessageJsonRpc from '../util/postmessage-json-rpc';
+
+/**
+ * @typedef {import('../../types/config').ConfigFromSidebar} ConfigFromSidebar
+ * @typedef {import('../../types/config').ConfigFromHost} ConfigFromHost
+ * @typedef {import('../../types/config').SidebarSettings} SidebarSettings
+ */
+
+/**
+ * Returns the global embedder ancestor frame.
+ *
+ * @param {number} levels - Number of ancestors levels to ascend.
+ * @param {Window=} window_
+ * @return {Window}
+ */
+function getAncestorFrame(levels, window_ = window) {
+  let ancestorWindow = window_;
+  for (let i = 0; i < levels; i++) {
+    if (ancestorWindow === ancestorWindow.top) {
+      throw new Error(
+        'The target parent frame has exceeded the ancestor tree. Try reducing the `requestConfigFromFrame.ancestorLevel` value in the `hypothesisConfig`'
+      );
+    }
+    ancestorWindow = ancestorWindow.parent;
+  }
+  return ancestorWindow;
+}
+
+/**
+ * Retrieve host configuration and list of groups from `targetFrame` via RPC
+ *
+ * @param {Window} targetFrame
+ * @param {string} origin - Origin filter for `window.postMessage` call
+ * @return {Promise<ConfigFromHost>}
+ */
+async function fetchConfigRPC(targetFrame, origin) {
```

Comment:
This function now only returns a `ConfigFromHost` — before it also merged it with `ConfigFromSidebar`, which is now part of `buildSettings`' remit.